modules/vnet_peering/main.tf

```diff
@@ -0,0 +1,31 @@
+data "azurerm_virtual_network" "local" {
```

Comment:
We do output virtual_network_id in the VNet module, so we could allow to either provide the IDs directly (thus skip extra datasource read) or get ones based on RG/VNet name as it's done right now.

Reply:
that would be nice, but what would happen if you specify both? we would need some additional logic to cover that. 

And that would be two additional variables, making this small module quite complex. I'm not saying no, I'm far from that, but maybe we should also rethink how to define variables so that they are more grouped? readable?

group them into maps? one for local vnet, one for remote? just and idea 🤔

Reply:
I refactored the code using maps - but it does seem that we have some checks failing for version 1.2 when the validate job is running . I believe the optional value inside the object is the issue :
```
  type = object({
    vnet_name                    = string
    resource_group_name          = string
    name                         = optional(string)
    allow_virtual_network_access = bool
    allow_forwarded_traffic      = bool
    allow_gateway_transit        = bool
    use_remote_gateways          = bool
  })
```

I did find something regarding this topic here : https://medium.com/@apr_1985/terraform-optional-attributes-and-defaults-d73643185341

Reply:
After discussion with @FoSix - we will use `map(any)` for now until we introduce a refactoring of the `map(any)` variables.